### ISSUE TITLE
ImageNet training pipeline applies the *test* transforms — no data augmentation

### ISSUE THREAD

**Issue by timoverse** [pwhofman/probly#559]
`_make_imagenet_loader` (`src/probly_benchmark/data.py`) maps
`TRANSFORMS_TEST["imagenet"]` (deterministic `Resize((224, 224))` + `Normalize`)
onto **all** loaders, including the train loader. ImageNet training therefore runs
without any data augmentation — no `RandomResizedCrop`, no `RandomHorizontalFlip` —
so every epoch feeds byte-identical images (only the order changes).

This deviates from the recipe the config cites (arXiv:2110.00476, "ResNet strikes
back", which relies on augmentation) and from the CIFAR-10 pipeline, which does
augment (`RandomCrop` + `RandomHorizontalFlip` in `get_data_train`).

**Affected:** all methods trained or fine-tuned on the `resnet50_imagenet` recipe,
including the subensemble/DARE backbone warmup paths.

**Proposed fix** (validated locally on the uint8-CHW tensors produced by the
`torchrgb8` decoder, then reverted):

```python
TRANSFORMS_TRAIN_IMAGENET = T.Compose([
    T.RandomResizedCrop((224, 224), antialias=True),
    T.RandomHorizontalFlip(),
    T.ToImage(),
    T.ToDtype(torch.float32, scale=True),
    T.Normalize((0.485, 0.456, 0.406), (0.229, 0.224, 0.225)),
])
```

Add a `transform` parameter to `_make_imagenet_loader` defaulting to the test
pipeline, and pass `TRANSFORMS_TRAIN_IMAGENET` only for the train loader in
`_get_imagenet_sharded`. Val/cal/test loaders must keep the deterministic test
pipeline.
